x/stake/client/rest/tx.go

```diff
@@ -0,0 +1,117 @@
+package rest
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/tendermint/go-crypto/keys"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/wire"
+	"github.com/cosmos/cosmos-sdk/x/stake"
+)
+
+func registerTxRoutes(ctx context.CoreContext, r *mux.Router, cdc *wire.Codec, kb keys.Keybase) {
+	r.HandleFunc("/stake/bondunbond", bondUnbondRequestHandlerFn(cdc, kb, ctx)).Methods("POST")
+}
+
+type bond struct {
+	Amount    sdk.Coin    `json:"amount"`
+	Candidate sdk.Address `json:"candidate"`
+}
```

Comment:
why are we recreating structures here??? we should be using the message types already provided in `x/stake/msg.go` in this circumstance it should be `type MsgDelegate struct`